pipeline/transformer/transformer_sierra/src/main/scala/weco/pipeline/transformer/sierra/transformers/SierraConceptIdentifier.scala

```diff
@@ -22,17 +23,63 @@ import weco.sierra.models.marc.VarField
 // https://www.loc.gov/marc/bibliographic/bd651.html
 // https://www.loc.gov/marc/bibliographic/bd655.html
 //
-object SierraConceptIdentifier {
+object SierraConceptIdentifier extends Logging {
 
-  def maybeFindIdentifier(varField: VarField,
-                          identifierSubfieldContent: String,
-                          ontologyType: String): Option[SourceIdentifier] = {
+  /**
+    * Determine the Library of Congress identifier type from the identifier value prefix.
+    *
+    * There are multiple LoC schemes that may be indicated by the same value in indicator2,
+    * The two we are interested in are LCSubjects and LCNames.  These can be differentiated by
+    * the first character in the identifier.
+    *
+    * Technically, the LoC schemes are differentiated by a sequence of alphabetic characters before
+    * the numbers start.
+    * - LC Names identifiers all have a one or two character prefix starting with n - e.g. n, no, nb.
+    * - LC Subjects identifiers all start with sh.
+    *
+    * There are other schemes that start with s (e.g. sj: Children's Subject Headings), but they are not
+    * in use in Wellcome data.
+    *
+    * In concise definition of these fields, (e.g. https://www.loc.gov/marc/bibliographic/concise/bd648.html)
+    * * a second indicator value of 0 indicates that the identifier comes from LCSH:
+    * 0 - Library of Congress Subject Headings
+    * However, the extended description (e.g. https://www.loc.gov/marc/bibliographic/bd648.html) goes on
+    * to also include the "Name authority files" (i.e. LCNames).
+    * 0 - Library of Congress Subject Headings
+    *  Subject added entry conforms to and is appropriate for use in the Library of Congress Subject Headings (LCSH) and the Name authority files that are maintained by the Library of Congress.
+    *
+    */
+  private def locScheme(idValue: String): IdentifierType =
+    idValue.split("\\d", 2).head match {
+      // sh is the only legal prefix for a Subject Headings identifier.
+      // At time of writing, there were some other s~ prefixed identifiers in use and marked as
+      // LCSH in the Works Catalogue, but these appear to be errors, so an exact match is
+      // enforced here in order to guard against such typos re-emerging.
```

Comment:
Incorrect prefixes I encountered include `shsh` `subjectsh` and `subject` (though I think the latter two may be from TEI)